src/openfermion/hamiltonians/_hubbard.py

```diff
@@ -117,36 +118,42 @@ def fermi_hubbard(x_dimension, y_dimension, tunneling, coulomb,
     Returns:
         hubbard_model: An instance of the FermionOperator class.
     """
+    tunneling = float(tunneling)
+    coulomb = float(coulomb)
+    chemical_potential = float(chemical_potential)
+    magnetic_field = float(magnetic_field)
+
     # Initialize fermion operator class.
     n_sites = x_dimension * y_dimension
-    if spinless:
-        n_spin_orbitals = n_sites
-    else:
-        n_spin_orbitals = 2 * n_sites
-    hubbard_model = FermionOperator((), 0.0)
-    # select particle-hole symmetry
+    n_spin_orbitals = n_sites
+
+    if not spinless:
+        n_spin_orbitals *= 2
+
+    hubbard_model = FermionOperator.zero()
+
+    # Select particle-hole symmetry
     if particle_hole_symmetry:
         coulomb_shift = FermionOperator((), 0.5)
     else:
-        coulomb_shift = FermionOperator((), 0.0)
+        coulomb_shift = FermionOperator.zero()
 
     # Loop through sites and add terms.
     for site in range(n_sites):
-
         # Add chemical potential and magnetic field terms.
         if chemical_potential and spinless:
-            coefficient = -1. * chemical_potential
+            coefficient = -chemical_potential
             hubbard_model += number_operator(
                 n_spin_orbitals, site, coefficient)
 
         if magnetic_field and spinless:
-            sign = (-1.) ** (site)
+            sign = -1 ** site
             coefficient = sign * magnetic_field
             hubbard_model += number_operator(
                 n_spin_orbitals, site, coefficient)
 
         if chemical_potential and not spinless:
```

Comment:
Since `chemical_potential` and `magnetic_field` now actually have a value (rather than None), you may want to consider replacing these 4 if statements (if chemical_potential and not spinless, etc.) with just 2 (if spinless, if not spinless). It would make it more readable.

Reply:
That could be nice, yeah :) The one thing I worry about is that it might lead to operators with zero coefficients which you'd want to compress - I'll look into it.

Reply:
A possibly good solution to that problem is to modify the init function of FermionOperator to always return a trivial zero operator if the coefficient passed to it is zero.

Reply:
Actually, what's the reason for this term more generally? There isn't any splitting with a magnetic field for the spinless case, right? I may just remove the field term there..

Reply:
I don't know, who defined the spinless Hubbard model in the first place?

Reply:
Looks like that code is fairly old, it must've been mistakenly added a while back. I think we can just agree to remove it then!